system/jlib/jbuff.cpp

```diff
@@ -736,6 +748,34 @@ const byte * MemoryBuffer::readDirect(size32_t len)
     return ret;
 }
 
+unsigned __int64 MemoryBuffer::readPacked()
+{
+    unsigned __int64 value = 0;
+    unsigned shift = 0;
+    loop
+    {
+        byte next;
+        read(next);
+        value = value | (((unsigned __int64)(next & 0x7f)) << shift);
+        if (!(next & 0x80))
+            break;
+        shift += 7;
+    }
+    return value;
+}
+
+MemoryBuffer &  MemoryBuffer::readPacked(unsigned & value)
+{
+    value = (unsigned)readPacked();
```

Comment:
Shouldn't this, in debug at least, impost a limit on number bytes read? i.e. if it reads >5 bytes it's overrun what could have possibly been serialized for a packed unsigned.